spyder/plugins/editor/widgets/tests/test_bookmarks.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Tests for bookmarks.
+"""
+
+# Standard library imports
+import os.path as osp
+
+try:
+    from unittest.mock import Mock
+except ImportError:
+    from mock import Mock  # Python 2
+
+# Third party imports
+import pytest
+from qtpy.QtGui import QTextCursor
+from qtpy.QtWidgets import QMainWindow
+
+# Local imports
+from spyder import version_info
+from spyder.py3compat import to_text_string
+from spyder.utils.qthelpers import qapplication
+import spyder.plugins.editor.widgets.codeeditor as codeeditor
+
+
+# --- Helper methods
+# -----------------------------------------------------------------------------
+def reset_emits(editor):
+    "Reset signal mocks."
+    editor.bookmarks_changed.reset_mock()
+
+
+def editor_assert_helper(editor, block=None, bm=None, emits=True):
+    """Run the tests for call to add_remove_breakpoint.
+
+    Args:
+        editor: CodeEditor instance.
+        block: Block of text.
+        bm: A list containing slots and columns of bookmarks
+        emits: Boolean to test if signals were emitted?
+    """
+    data = block.userData()
+    assert data.bookmarks == bm
+    if emits:
+        editor.bookmarks_changed.emit.assert_called_with()
+    else:
+        editor.bookmarks_changed.emit.assert_not_called()
+
+
+# --- Fixtures
+# -----------------------------------------------------------------------------
+@pytest.fixture
+def code_editor_bot(qtbot):
+    """Create code editor with default Python code."""
+    editor = codeeditor.CodeEditor(parent=None)
+    indent_chars = ' ' * 4
+    tab_stop_width_spaces = 4
+    editor.setup_editor(language='Python', indent_chars=indent_chars,
+                        tab_stop_width_spaces=tab_stop_width_spaces)
+    # Mock the signal emit to test when it's been called.
+    editor.bookmarks_changed = Mock()
+    text = ('def f1(a, b):\n'
+            '"Double quote string."\n'
+            '\n'  # Blank line.
+            '    c = a * b\n'
+            '    return c\n'
+            )
+    editor.set_text(text)
+    return editor, qtbot
+
+
+# ---- Qt Test Fixtures
+@pytest.fixture
+def setup_editor(qtbot, monkeypatch):
+    """Set up the Editor plugin."""
+    qapplication()
+    from spyder.plugins.editor.plugin import Editor
+
+    monkeypatch.setattr('spyder.plugins.editor.plugin.add_actions', Mock())
+
+    class MainMock(QMainWindow):
+        def __getattr__(self, attr):
+            if attr.endswith('actions'):
+                return []
+            elif attr == 'projects':
+                projects = Mock()
+                projects.get_active_project.return_value = None
+                return projects
+            else:
+                return Mock()
+
+    window = MainMock()
+    editor = Editor(window)
+    window.setCentralWidget(editor)
+    window.resize(640, 480)
+    qtbot.addWidget(window)
+    window.show()
+
+    yield editor
+
```

Comment:
Missing the ``editor.close()`` after the ``yield`` (as the fixture you copied had); this would be resolved doing the above.

Reply:
I can't find this, where does this ```editor.close()``` come from?